rts/lra/lra-test/lra-test-arquillian-extension/src/main/java/io/narayana/lra/arquillian/spi/NarayanaLraRecovery.java

```diff
@@ -0,0 +1,64 @@
+package io.narayana.lra.arquillian.spi;
+
+import io.narayana.lra.LRAConstants;
+import io.narayana.lra.client.NarayanaLRAClient;
+import org.eclipse.microprofile.lra.tck.service.spi.LraRecoveryService;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+
+@ApplicationScoped
+public class NarayanaLraRecovery implements LraRecoveryService {
+
+    @Override
+    public void triggerRecovery() {
+        invokeRecovery(null, System.getProperty(NarayanaLRAClient.LRA_COORDINATOR_HOST_KEY, "localhost"),
+            Integer.parseInt(System.getProperty(NarayanaLRAClient.LRA_COORDINATOR_PORT_KEY, "8080")));
+    }
+
+    @Override
+    public void triggerRecovery(URI lraId) {
+        invokeRecovery(lraId, lraId.getHost(), lraId.getPort());
+    }
+
+    private void invokeRecovery(URI lraId, String host, int port) {
+        if (invokeRecoveryCoordinatorEndpoint(lraId, host, port)) {
+            // first recovery scan probably collided with periodic recovevery which started
+            // before the test execution so try once more
+            invokeRecoveryCoordinatorEndpoint(null, host, port);
```

Comment:
Why is the first invocation made with a non null LRA id and the second one uses a null one. I think both invocations need the LRA id to be passed in.